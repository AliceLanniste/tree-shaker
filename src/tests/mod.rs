fn tree_shake(input: &str) {
  let result = crate::tree_shake(input);
  let output = result.codegen_return.source_text;
  insta::assert_snapshot!(output);
}

#[test]
fn test_1() {
  tree_shake(
    r#"
      export let a = 1 && 2;
      export { b };
      let c = 3;
      let { ["b"]: b, d } = { b: 2, d: effect };
      "#,
  );
}
