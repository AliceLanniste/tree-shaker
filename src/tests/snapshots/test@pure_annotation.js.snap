---
source: src/tests/mod.rs
expression: tree_shake(input)
input_file: src/tests/fixtures/pure_annotation.js
---
other1();
other2();
other4();
export const a = unknown4.g?.(1, other5());
function simple() {
	return a;
}
other1();
export const b = /*#__PURE__*/ simple(other2());
export const c = /*#__PURE__*/ simple();
function mutate(obj) {
	obj.a = b;
	delete obj.g;
	return obj.f();
}
export const d = /*#__PURE__*/ mutate({ f: effect });
